src/Calculations.js

```diff
@@ -45,11 +47,20 @@ const isOutlier = (value, limit = 3.5) => {
 // it is from a sample. This is important for calculating sample
 // standard deviation
 // will return 0 if array is empty
-const mean = (values, { sample = 0, conversionFactor = 1 } = {}) => {
+const mean = (
+  values,
+  { sample = 0, conversionExpression = `(${originalValue})` } = {}
+) => {
+  if (conversionExpression.indexOf(originalValue) === -1) {
+    return NaN
+  }
+
   let sum = values.reduce(function (a, b) {
     return a + b
   }, 0)
-  return (sum / (values.length - sample) || 0) * conversionFactor
+  let mean = sum / (values.length - sample) || 0
```

Comment:
Thanks for the reminder.  I was thinking that `let` means not mutated because that's the case in the Swift language, where `var` is a mutatable variable.  I remember now that JavaScript has legacy `var`, now `let` for mutatable variables and `const` which is what I should have used here.  The whole code base is littered with `let`s that don't get mutated.  Maybe that's something we could fix with a linter.

Reply:
Agreed. Not asking you to change them all. Quanthub was done a long time ago so I wasn't aware of the standards. In traction I have been a lot more strict. Not sure if you can use Lint. Interestingly we have been talking about trying out github copilot so I think it might be able to do something like that. Also introducing typescript can have a benefit in this area.

Reply:
Auto-fixed by ESLint: https://github.com/sanger/quanthub/pull/210